codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -333,12 +333,36 @@ private static void validateChar(char character) {
         case '=':
             throw new IllegalArgumentException(
                "a header name cannot contain the following prohibited characters: =,;: \\t\\r\\n\\v\\f: " +
-                       character);
+                       value);
         default:
             // Check to see if the character is not an ASCII character, or invalid
-            if (character > 127) {
+            if (value < 0) {
                 throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " +
-                        character);
+                        value);
+            }
+        }
+    }
+
+    private static void validateHeaderNameElement(char value) {
```

Comment:
@normanmaurer - I broke this out for testing to avoid having to `1. cast to int. 2. mask 0xFF 3. cast to char` for each `byte` if AsciiString is used.  I haven't yet verified if this is justified yet...